cmd/clusterctl/examples/vsphere/machineset.yaml.template

```diff
@@ -15,30 +15,25 @@ spec:
         cluster.k8s.io/cluster-name: ${CLUSTER_NAME}
     spec:
       providerSpec:
-        value:
-          apiVersion: "vsphereproviderconfig/v1alpha1"
-          kind: "VsphereMachineProviderConfig"
-          machineSpec:
-            datacenter: ""
-            datastore: ""
-            resourcePool: ""
-            networks:
-            - networkName: ""
-              ipConfig:
-                networkType: dhcp
-            numCPUs: 2
-            memoryMB: 2048
-            template: ""
-            disks:
-            - diskLabel: ""
-              diskSizeGB: 20
-            preloaded: false
-            trustedCerts:
-            - zzzz
-            - aaaa
-            ntpServers:
-            - 0.pool.ntp.org
-            - 1.pool.ntp.org
+	value:
+	  apiVersion: "vsphereproviderconfig/v1alpha1"
+	  kind: "VsphereMachineProviderConfig"
+	  machineSpec:
+	    datacenter: "${VSPHERE_DATACENTER}"
+	    datastore: "${VSPHERE_DATASTORE}"
+	    resourcePool: "${VSPHERE_RESOURCE_POOL}"
+	    vmFolder: "${VSPHERE_FOLDER}"
+	    networks:
+	    - networkName: "${VSPHERE_NETWORK}"
+	      ipConfig:
+		networkType: "${VSPHERE_NETWORK_TYPE}"
+	    numCPUs: 2
+	    memoryMB: 2048
+	    template: "${VSPHERE_TEMPLATE}"
+	    disks:
+	    - diskLabel: "${VSPHERE_DISK}"
+	      diskSizeGB: ${VSPHERE_DISK_SIZE_GB}
+	    preloaded: false
       versions:
         kubelet: 1.11.3
       roles:
```

Comment:
Can we remove the rolese section. It is not used is just a leftover from older design.
```
roles:
.....
```

Reply:
That seems like a significant change because some of the CAPV logic still depends on the roles being there (at least last time I checked). Either way sounds like something I can address in a follow-up PR :)

Reply:
Hi @sidharthsurana,

I personally would like more information on the RBAC anyway as CAPA seems to generate their RBAC manifests through the controller gen tool, and it looks like we intended to do that at one point but didn't get that far.

Reply:
This has nothing to do with RBAC. This is a just yet another API field in the spec, that IIRC pre-dates late 2018
I am surprised CAPA uses that since the method in CAPI that checks whether a machine is a control plane machine does depend on this

Reply:
> That seems like a significant change because some of the CAPV logic still depends on the roles being there (at least last time I checked). Either way sounds like something I can address in a follow-up PR :)
@andrewsykim
I don't recall any CAPV logic depending on that. Can you point me to one? maybe I am mistaken.

Reply:
Ah, I was mistakening this for versions :). Either way, don't want to mix up those changes in this PR, will update in follow- up